specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1080 @@
+- Through the operation of the subnet, the `CollateralThreshold` may change according to the `CurrGasRate` reported by the subnet to the parent (through the commitment of checkpoints), and the `MisbehaviorsLevels`, a measure of the number of successful fault proofs for the child subnet committed in the parent, requiring validators to update their collateral accordingly. 
+
+### Detectable misbehaviors
+In the following table we specify a list of detectable misbehaviors that we envisions users to be able to report through the commitment of fault proofs, along with the `impact` of the attack. This `impact` parameter is used to determine the amount from the collateral slashed due to the reported misbehavior.
+
+| Misbehavior         | Impact     | Description | Fault Proof |
+|--------------|-----------|------------|------------|
+| (R4.1) Agreement equivocation               |1|  Deviation on consensus (votes on consensus, checks PoS-based, etc.) | Block checks |
+| (R4.2) Invalid state transition             |2| Consensus reaches valid blocks with an invalid state transition   | State replay  |
+
+- __Agreement equivocation__: A malicious majority in a subnet can deviate from the consensus algorithm and propagate as valid a block that should have been considered invalid if the protocol was followed successfully. Agreement equivocations may arise, for instance, in blocks that do not pass all block checks (e.g. blocks are not mined in a different epoch, the block does not includes the right ticket, the block is not signed by the right leader, etc.); if the block doesn't include the right votes in a BFT-like consensus, etc.
+To report an agreement equivocation, users need to submit to the parent's SCA the chain of blocks including the equivocated block, and the valid chain replacing the invalid block with a valid one. The SCA perfoms syntactic checks over the inputs and runs the `equivocation_check` function in the corresponding actor of the subnet actor to perform deeper consensus-specific checks. If any of these checks fail, the collateral of the subnet is slashed. The function `equivocation_check` is a mandatory function in the subent actor interface that implements all the consensus-specific checks required to detect agreement equivocations.
+Thus, if an agreement equivocation is detected in epoch `n`, a user looking to report the misbehavior needs to report the chain of the latest `n-finality` including the equivocated block, and the valid chain of `n-finality` blocks to the parent SCA. For BFT-like protocols, it may suffice to report the latest block where the equivocation arises, while for sync protocols like Nakamoto consensus --without a clear metric of finality and where long-lasting forks may appear-- reporting agreement equivocations may be impossible.
```

Comment:
```suggestion
Thus, if an agreement equivocation is detected in epoch `n`, a user looking to report the misbehavior needs to report the chain of the latest `n-finality` including the equivocated block and the valid chain of `n-finality` blocks to the parent SCA. For BFT-like protocols, it may suffice to report the latest block where the equivocation arises, while for sync protocols like Nakamoto consensus -- without a clear metric of finality and where long-lasting forks may appear -- reporting agreement equivocations may be impossible.
```